###Lukas Mosser

##Derivation of Spherical symmetrical diffusivity equation

$$Mass \ In = A(r)\rho(r)u(r)\Delta t $$
$$Mass \ In = A(r+\Delta r)\rho(r + \Delta r)u(r+ \Delta r)\Delta t $$
$$Accumulation = m(t+dt)-m(t)$$

We divide by $\Delta t$ and let the limit of time go towards 0:
$$\lim_{\Delta t \to 0} \frac{m(t+dt)-m(t)}{\Delta t}=\frac{dm}{dt}$$

We define the mass as:
$$m=V \rho \phi$$
Where the Volume of the differential Sphere can be derived from:
$$V = V(r+\Delta r)_{Sphere} - V(r)_{Sphere} $$

Where:
    $$V(r)_{Sphere}=\frac{4}{3}\pi r^3$$
and:
    $$V(r+\Delta r)_{Sphere}=\frac{4}{3}\pi (r + \Delta r)^3$$
respectively.

The volume of the spherical control volume is then given by:
$$V = \frac{4}{3}\pi (r + \Delta r)^3 - \frac{4}{3}\pi r^3$$
Expanding $(r + \Delta r)^3$ leads to the volume of the spherical control volume:
$$V = 4\pi r^2\Delta r$$

The mass of the spherical control volume is therefore given by:
    $$m=4\pi r^2\Delta r\rho\phi$$

We can rewrite the accumulation in terms of the spherical control volumes mass:
$$\frac{dm}{dt} = \frac{\partial 4\pi r^2\Delta r*\rho*\phi}{\partial t} = 4\pi r^2\Delta r \frac{\partial (\rho \phi)}{\partial t}$$

We apply the continuity equation:
$$Mass In = Mass Out \ + \ Accumulation$$
Forming LHS Terms:
$$Mass In \ - \ Mass Out  = Accumulation$$

We define Mass In in terms of the spherical control volume ($\Delta t$ has been rearranged to form the limit of the accumulation):
    $$Mass \ In = 4\pi r^2 \rho(r)u(r)$$

We define Mass Out in terms of the spherical control volume ($\Delta t$ has been rearranged to form the limit of the accumulation):
    $$Mass \ Out = 4\pi (r+\Delta r)^2 \rho(r+\Delta r)u(r+\Delta r)$$

LHS terms can be rewritten using above definitions:
$$LHS = 4\pi r^2 \rho(r)u(r) - 4\pi (r+\Delta r)^2 \rho(r+\Delta r)u(r+\Delta r) $$

Rearranging the left hand side and dividing LHS and RHS by $\Delta r$:
$$LHS = - \frac{(4\pi (r+\Delta r)^2 \rho(r+\Delta r)u(r+\Delta r) - 4\pi r^2 \rho(r)u(r))}{\Delta r}$$

Performing the limit as $\Delta r$ goes towards 0:
$$LHS = -\lim_{\Delta r \to 0}\frac{(4\pi (r+\Delta r)^2 \rho(r+\Delta r)u(r+\Delta r) - 4\pi r^2 \rho(r)u(r))}{\Delta r}$$
$$LHS = -\frac{\partial (4\pi r^2 \rho u)}{\partial r}$$
Rearranging constant terms gives us the final expression for the LHS:
$$LHS = -4\pi\frac{\partial (r^2 \rho u)}{\partial r}$$

Combining LHS and RHS terms:
$$ LHS = RHS $$
$$-4\pi\frac{\partial (r^2 \rho u)}{\partial r} =  4\pi r^2 \frac{\partial (\rho \phi)}{\partial t}$$
Gives the continuity equation for a spherical symmetrical coordinate system:
$$-\frac{\partial (r^2 \rho u)}{\partial r} =  r^2 \frac{\partial (\rho \phi)}{\partial t}$$

Writing Darcy's equation in 1D Spherical coordinates:
$$ u = -\frac{k}{\mu}\frac{\partial P}{\partial r}$$
And Equations of State for the fluid and solid phase respectively:
$$c_{o} = \frac{1}{\rho}\frac{\partial \rho}{\partial P}$$
$$c_{f} = \frac{1}{\phi}\frac{\partial \phi}{\partial P}$$

Using Darcy's equation we can rewrite the continuity equations LHS:
$$ LHS = -\frac{\partial (r^2 \rho u)}{\partial r} = \frac{\partial \ }{\partial r}(r^2 \rho \frac{k}{\mu}\frac{\partial P}{\partial r}) = \frac{k}{\mu} \frac{\partial \ }{\partial r}(r^2 \rho \frac{\partial P}{\partial r})$$

Applying the Chain Rule to the LHS:
$$LHS = \frac{k}{\mu}(2r \rho \frac{\partial P}{\partial r} + r^2 \rho \frac{\partial^2 P}{\partial r^2} + r^2 \frac{\partial \rho}{\partial r} \frac{\partial P}{\partial r})$$
$$LHS = \frac{k}{\mu}(2r \rho \frac{\partial P}{\partial r} + r^2 \rho \frac{\partial^2 P}{\partial r^2} + r^2 \frac{\partial \rho}{\partial r} \frac{\partial P}{\partial P} \frac{\partial P}{\partial r})$$
$$LHS = \frac{k}{\mu}(2r \rho \frac{\partial P}{\partial r} + r^2 \rho \frac{\partial^2 P}{\partial r^2} + r^2 \frac{\partial \rho}{\partial P} \frac{\partial P}{\partial r} \frac{\partial P}{\partial r})$$
$$LHS = \frac{k}{\mu}(2r \rho \frac{\partial P}{\partial r} + r^2 \rho \frac{\partial^2 P}{\partial r^2} + r^2 \frac{\partial \rho}{\partial P} (\frac{\partial P}{\partial r})^2)$$
$$LHS = \rho \frac{k}{\mu}(2r \frac{\partial P}{\partial r} + r^2 \frac{\partial^2 P}{\partial r^2} + r^2 \frac{1}{\rho} \frac{\partial \rho}{\partial P} (\frac{\partial P}{\partial r})^2)$$
$$LHS = \rho \frac{k}{\mu}(2r \frac{\partial P}{\partial r} + r^2 \frac{\partial^2 P}{\partial r^2} + r^2 c_{o} (\frac{\partial P}{\partial r})^2)$$

Neglecting small terms $(\frac{\partial P}{\partial r})^2$ gives final form of LHS
$$LHS = \rho \frac{k}{\mu}(2r \frac{\partial P}{\partial r} + r^2 \frac{\partial^2 P}{\partial r^2})$$
$$LHS = \rho \frac{k}{\mu}\frac{\partial \ }{\partial r}(r^2 \frac{\partial P}{\partial r})$$

Considering the RHS and expanding using the Chain rule:
$$RHS = r^2 \frac{\partial (\rho \phi)}{\partial t} = r^2 (\phi \frac{\partial \rho}{\partial t} + \rho\frac{\partial \phi }{\partial t})$$
Rearranging Terms and expanding by $\frac{\partial P}{\partial P}$:
$$RHS = r^2 \rho \phi (\frac{1}{\rho} \frac{\partial P}{\partial P}\frac{\partial \rho}{\partial t} + \frac{1}{\phi}\frac{\partial P}{\partial P}\frac{\partial \phi }{\partial t})$$
$$RHS = r^2 \rho \phi (\frac{1}{\rho} \frac{\partial \rho}{\partial P}\frac{\partial P}{\partial t} + \frac{1}{\phi}\frac{\partial \phi}{\partial P}\frac{\partial P }{\partial t})$$

Using the equations of state we can rearrange the RHS:
$$RHS = r^2 \rho \phi (c_{o} + c_{f})\frac{\partial P}{\partial t}$$
We define the total system compressibility (single fluid) as $c_{t}=c_{o}+c_{f}$
$$RHS = r^2 \rho \phi c_{t}\frac{\partial P}{\partial t}$$

Combining LHS and RHS:
$$\rho \frac{k}{\mu}\frac{\partial \ }{\partial r}(r^2 \frac{\partial P}{\partial r}) = r^2 \rho \phi c_{t}\frac{\partial P}{\partial t}$$

Simplyfing we arrive at the final pressure diffusion equation for spherical symmetrical flow:
    $$\frac{1}{r^2}\frac{\partial \ }{\partial r}(r^2 \frac{\partial P}{\partial r}) =  \frac{\mu \phi c_{t}}{k} \frac{\partial P}{\partial t}$$

##Dimensionless form of the spherical symmetrical diffusion equation

We define dimensionless groups for space:
$$r_{D} = \frac{r}{r_{w}}$$

$$\frac{r_{w}^3}{r_{w}^3}\frac{1}{r^2}\frac{\partial \ }{\partial r}(r^2 \frac{\partial P}{\partial r}) =  \frac{\mu \phi c_{t}}{k} \frac{\partial P}{\partial t}$$

$$\frac{1}{r_{w}^3} \frac{1}{r_{D}^2}\frac{\partial }{\partial r_{D}}(r^2 \frac{\partial P}{\partial r}) =  \frac{\mu \phi c_{t}}{k} \frac{\partial P}{\partial t}$$

$$\frac{1}{r_{w}} \frac{1}{r_{D}^2}\frac{\partial }{\partial r_{D}}(r_{D}^2 \frac{r_{w}}{r_{w}}\frac{\partial P}{\partial r}) =  \frac{\mu \phi c_{t}}{k} \frac{\partial P}{\partial t}$$

$$\frac{1}{r_{w}} \frac{1}{r_{D}^2}\frac{\partial }{\partial r_{D}}(r_{D}^2 \frac{1}{r_{w}}\frac{\partial P}{\partial r_{D}}) =  \frac{\mu \phi c_{t}}{k} \frac{\partial P}{\partial t}$$

$$\frac{1}{r_{w}^2} \frac{1}{r_{D}^2}\frac{\partial }{\partial r_{D}}(r_{D}^2 \frac{\partial P}{\partial r_{D}}) =  \frac{\mu \phi c_{t}}{k} \frac{\partial P}{\partial t}$$

$$\frac{1}{r_{D}^2}\frac{\partial }{\partial r_{D}}(r_{D}^2 \frac{\partial P}{\partial r_{D}}) =  \frac{r_{w}^2 \mu \phi c_{t}}{k} \frac{\partial P}{\partial t}$$

We define dimensionless time as:
    $$t_{D} = \frac{k}{r_{w}^2 \mu \phi c_{t}}t $$
And perform a change of variables from time to dimensionless time:
$$\frac{1}{r_{D}^2}\frac{\partial }{\partial r_{D}}(r_{D}^2 \frac{\partial P}{\partial r_{D}}) =  \frac{\partial P}{\partial t_{D}}$$

We define dimensionless pressure:
    $$P_{D}=\frac{2 \pi kh}{\mu q} P $$
And perform a change of varibles from pressure to dimensionless pressure:
$$\frac{1}{r_{D}^2}\frac{\partial }{\partial r_{D}}(r_{D}^2 \frac{\partial P}{\partial r_{D}}) \frac{2 \pi kh}{\mu q}=  \frac{\partial P}{\partial t_{D}}\frac{2 \pi kh}{\mu q}$$
$$\frac{1}{r_{D}^2}\frac{\partial }{\partial r_{D}}(r_{D}^2 \frac{\partial P_{D}}{\partial r_{D}})=  \frac{\partial P_{D}}{\partial t_{D}}$$

The diffusivity equation for spherical symmetrical flow in dimensionless terms is therefore given as:
$$\frac{1}{r_{D}^2}\frac{\partial }{\partial r_{D}}(r_{D}^2 \frac{\partial P_{D}}{\partial r_{D}})=  \frac{\partial P_{D}}{\partial t_{D}}$$
With dimensionless groups:
$$P_{D}=\frac{2\pi kh}{\mu q} (P_{i}-P) $$
$$t_{D} = \frac{k}{r_{w}^2 \mu \phi c_{t}}t $$
$$r_{D} = \frac{r}{r_{w}}$$

##Dimensionless groups for spherical symmetrical flow in field units:

In consistent units we define the dimensionless groups for the diffusivity equation in spherical symmetrical coordinates as:
$$P_{D}=\frac{2\pi kh}{\mu q}  (P_{i}-P) $$
$$t_{D} = \frac{k}{r_{w}^2 \mu \phi c_{t}}t $$
$$r_{D} = \frac{r}{r_{w}}$$

Following conversion factors are given from CGS units (consistent) to field units (not consistent):
$$k = 1 [D] =  1000 [mD] $$
$$\mu = 1 [cP] = 1 [cP]$$
$$c_{t}= 1 [\frac{1}{atm}] = \frac{1}{14.6959} [\frac{1}{psia}]$$
$$q = 1.84 [\frac{cm^3}{s}]= 1 [\frac{BBl}{d}] $$
$$h = 30.48 [cm] = 1 [ft]$$
$$B = 1 [-] = 1 [-]$$
$$3600 [s] = 1 [h]$$

We can therefore express the dimensionless variables in field units:
    

In [33]:
P_D = 1/(2*3.141)*1000*1/30.48*1.84*14.6959
print "Conversion factor P_D:", P_D

t_D = 1000/((1/30.48)**2*1/14.6959)*1/(60*60)
print "Conversion factor t_D:", 1/t_D

Conversion factor P_D: 141.221596345
Conversion factor t_D: 0.000263679512654


We can therefore express the dimensionless groups in EOF units as:
$$P_{D}= \frac{k [mD]h[ft]}{141.2 \mu[cP] q[\frac{BBl}{d}]} (P_{i}-P)[psia] $$
$$t_{D} = \frac{0.00026367k[mD]}{ r_{w}^2 [ft]^2 \mu [cP] \phi [-] c_{t}[\frac{1}{psia}]B[\frac{rb}{STB}]}t [h]$$
$$r_{D} = \frac{r[ft]}{r_{w}[ft]}$$